packages/app/pages/checkout/[network]/[address].tsx

```diff
@@ -4,21 +4,45 @@ import {
   contractAddressesByNetworkId,
   getAbiFromJson,
   networkIdByName,
+  firebaseConfig,
 } from "../../../components/constants";
 import { ContractSendMethod } from "web3-eth-contract";
 import Web3 from "web3";
 import passportFactoryJson from "@cabindao/nft-passport-contracts/artifacts/contracts/PassportFactory.sol/PassportFactory.json";
 import passportJson from "@cabindao/nft-passport-contracts/artifacts/contracts/Passport.sol/Passport.json";
 import { styled } from "../../../stitches.config";
 import { Button } from "@cabindao/topo";
-import { useCallback, useRef, useState } from "react";
 import BN from "bn.js";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { initializeApp } from "firebase/app";
+import { 
+  getFirestore, 
+  collection, 
+  doc, 
+  getDoc 
+} from 'firebase/firestore/lite';
 
 type QueryParams = {
   network: string;
   address: string;
 };
 
+// Create a connection to the firebase DB.
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
+
+// Method to check redirection URL for current passport.
+async function getRedirectionUrl(db: any, address: string) {
+  const urlCol = collection(db, 'redirect-urls');
+  const membershipDoc = doc(urlCol, address)
+  const membership = await getDoc(membershipDoc);
+  const membershipData = membership.data();
+  if (membershipData && membershipData["redirect_url"]) {
+    return membershipData["redirect_url"];
+  }
+  return "";
```

Comment:
nit: you could use optional chaining to make this one line

`return membershipData?.["redirect_url"] || ""`

Reply:
done